File: Planarian/Planarian/Modules/Leads/Services/LeadService.cs
using Planarian.Model.Database.Entities.Leads;
using Planarian.Model.Shared;
using Planarian.Modules.Leads.Repositories;
using Planarian.Shared.Base;

namespace Planarian.Modules.Leads.Services;

public class LeadService : ServiceBase<LeadRepository>
{
    public LeadService(LeadRepository repository, RequestUser requestUser) : base(repository, requestUser)
    {
    }

    public async Task<LeadVm?> GetLeads(string leadId)
    {
        var lead = await Repository.GetLead(leadId);
        return lead;
    }

    public async Task DeleteLead(string leadId)
    {
        var lead = await Repository.Get(leadId);

        if (lead == null) throw new ArgumentOutOfRangeException(nameof(lead), "Lead not found");
        Repository.Delete(lead);

        await Repository.SaveChangesAsync();
    }
}

File: Planarian.Web/src/Modules/Lead/Services/LeadService.ts
import { HttpClient } from "../../..";

const baseUrl = "api/leads";
const LeadService = {
  //#region Leads

  async DeleteLead(leadId: string): Promise<void> {
    const response = await HttpClient.delete<void>(`${baseUrl}/${leadId}`);
  },

  //#endregion
};
export { LeadService };
